packages/backend/src/accounting/psql/ledger-transfer/index.ts

```diff
@@ -38,3 +65,135 @@ export async function getAccountTransfers(
     { credits: [], debits: [] } as GetTransfersResult
   )
 }
+
+export async function createTransfers(
+  deps: ServiceDependencies,
+  transfers: CreateTransferArgs[],
+  trx?: TransactionOrKnex
+): Promise<CreateTransfersResult> {
+  const validationResults = await Promise.all(
+    transfers.map((transfer) =>
+      validateTransfer(deps, {
+        creditAccount: transfer.creditAccount,
+        debitAccount: transfer.debitAccount,
+        amount: transfer.amount
+      })
+    )
+  )
+
+  const errors: CreateTransferError[] = []
+
+  for (const [i, maybeError] of validationResults.entries()) {
+    if (isTransferError(maybeError)) {
+      errors.push({
+        index: i,
+        error: maybeError
+      })
+    }
+  }
+
+  if (errors.length > 0) {
+    return {
+      results: [],
+      errors
+    }
+  }
+
+  try {
+    const createdTransfers = await LedgerTransfer.query(
+      trx || deps.knex
+    ).insertAndFetch(transfers.map(prepareTransfer))
+
+    return {
+      results: createdTransfers,
+      errors: []
+    }
+  } catch (error) {
+    if (error instanceof UniqueViolationError) {
+      return {
+        results: [],
+        errors: [{ index: 0, error: TransferError.TransferExists }]
+      }
```

Comment:
We _could_ add a proper index for the `TransferError.TransferExists` error, by parsing the `UniqueViolationError` details to get the `transferRef` that violated the constraint, but I felt like it's not necessary (yet)

Reply:
What about doing `index: -1` to maybe avoid confusion?

Reply:
Yup, makes sense to me